src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -1,49 +1,79 @@
-import React, {
-  Component,
-  ReactNode,
-  ReactElement,
-  forwardRef,
-  useRef,
-} from 'react';
+import React, { Component, ReactNode, ReactElement, forwardRef } from 'react';
 import { default as styled } from 'styled-components';
 import classnames from 'classnames';
-import {
-  ListboxInput,
-  ListboxButton,
-  ListboxButtonProps,
-  ListboxPopoverProps,
-  ListboxList,
-  ListboxPopover,
-  useListboxContext,
-} from '@reach/listbox';
-import { positionMatchWidth } from '@reach/popover';
 import { getConditionalAriaProp } from '../../../utils/aria';
-import { useEnhancedEffect } from '../../../utils/common';
 import { getLogger } from '../../../utils/log';
 import { AutoId } from '../../utils/AutoId/AutoId';
-import { HtmlSpan, HtmlDiv } from '../../../reset';
+import { HtmlSpan, HtmlDiv, HtmlInput } from '../../../reset';
 import { Label, LabelMode } from '../../Form/Label/Label';
 import { DropdownItemProps } from '../DropdownItem/DropdownItem';
 import { baseStyles } from './Dropdown.baseStyles';
 import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
+import {
+  forkRefs,
+  getOwnerDocument,
+  getRecursiveChildText,
+} from '../../../utils/common/common';
+import { Popover } from '../../../core/Popover/Popover';
+import { SelectItemList } from '../../Form/Select/BaseSelect/SelectItemList/SelectItemList';
+import { HintText } from '../../Form/HintText/HintText';
+import { StatusText } from '../../Form/StatusText/StatusText';
+import { StatusTextCommonProps } from '../../Form/types';
+
+// Dropdown is a button but it also has a listbox popup
+/* eslint-disable jsx-a11y/role-supports-aria-props */
```

Comment:
Maybe this disable should be placed more specifically to where the rule is disabled, only for that place, if possible

Reply:
There are multiple props this this affects so I thought it best to just put at the top of the file

Reply:
Is it a bit of "code smell" if some aria props are not compatible element's role?

Reply:
Ha, making the element a <HtmlButton> and removing role="button" made the linter happy with the props